src/app/plan/duck/selectors.ts

```diff
@@ -102,134 +103,54 @@ const getPlansWithPlanStatus = createSelector(
-      }
-
-      hasClosedCondition = !!plan.MigPlan.status.conditions.filter((c) => c.type === 'Closed')
-        .length;
-      hasReadyCondition = !!plan.MigPlan.status.conditions.filter((c) => c.type === 'Ready').length;
-      hasPlanError = plan.MigPlan.status.conditions.filter((c) => c.category === 'Critical') > 0;
-      hasConflictCondition = !!plan.MigPlan.status.conditions.some(
-        (c) => c.type === 'PlanConflict'
-      );
-      hasPODWarnCondition = !!plan.MigPlan.status.conditions.some(
-        (c) => c.type === 'PodLimitExceeded'
+      const isPlanLocked = !!lockedPlans.some(
+        (lockedPlan) => lockedPlan === plan.MigPlan.metadata.name
       );
-      hasPVWarnCondition = !!plan.MigPlan.status.conditions.some(
-        (c) => c.type === 'PVLimitExceeded'
-      );
-
-      const planConflictCond = plan.MigPlan.status.conditions.find(
-        (c) => c.type === 'PlanConflict'
-      );
-      if (planConflictCond) {
-        conflictErrorMsg = planConflictCond.message;
-      }
-
-      if (plan.Migrations.length) {
-        const latest = plan.Migrations[0];
-
-        hasPrevMigrations = !!plan.Migrations.length;
-        latestType = latest.spec.stage ? 'Stage' : 'Migration';
-
-        if (latest.status && latest.status.conditions) {
-          latestIsFailed = !!latest.status.conditions.some((c) => c.type === 'Failed');
-          hasCancelingCondition = !!latest.status.conditions.some((c) => c.type === 'Canceling');
-          hasCanceledCondition = !!latest.status.conditions.some((c) => c.type === 'Canceled');
+      const latestMigration = plan.Migrations.length ? plan.Migrations[0] : null;
+      const latestType = latestMigration?.spec?.stage ? 'Stage' : 'Migration';
+
+      const hasSucceededStage = !!plan.Migrations.filter((m) => {
+        if (m.status && m.spec.stage) {
+          return (
+            m.status.conditions.some((c) => c.type === 'Succeeded') &&
+            !m.status.conditions.some((c) => c.type === 'Canceled')
+          );
         }
+      }).length;
 
-        hasSucceededStage = !!plan.Migrations.filter((m) => {
-          if (m.status && m.spec.stage) {
-            return (
-              m.status.conditions.some((c) => c.type === 'Succeeded') &&
-              !m.status.conditions.some((c) => c.type === 'Canceled')
-            );
-          }
-        }).length;
-
-        hasSucceededMigration = !!plan.Migrations.filter((m) => {
-          if (m.status && !m.spec.stage) {
-            return (
-              m.status.conditions.some((c) => c.type === 'Succeeded') &&
-              !latest.status.conditions.some((c) => c.type === 'Canceled')
-            );
-          }
-        }).length;
+      const hasSucceededMigration = !!plan.Migrations.filter((m) => {
+        if (m.status && !m.spec.stage) {
+          return (
+            m.status.conditions.some((c) => c.type === 'Succeeded') &&
+            !latestMigration.status.conditions.some((c) => c.type === 'Canceled')
```

Comment:
I know this was already present in the existing logic, but I just noticed that `hasSucceededMigration` is checking for a canceled condition on only the latest migration here every time, but above in `hasSucceededStage` it checks for a canceled condition on the current migration being checked by filter predicate. Is that intentional? Not sure I fully grok the logic.

Reply:
Good question - the issue is that on Cancelled migrations/stages, there is sometimes a succeeded condition also.